x-pack/platform/plugins/shared/fleet/server/services/epm/elasticsearch/template/template.test.ts

```diff
@@ -55,380 +68,447 @@ describe('EPM template', () => {
 
-    const processedFields = processFields(fields);
-    const mappings = generateMappings(processedFields);
+      const templateWithDataStreamHidden = getTemplate({
+        templateIndexPattern,
+        type: 'logs',
+        packageName: 'nginx',
+        composedOfTemplates: [],
+        templatePriority: 200,
+        registryElasticsearch: {
+          'index_template.data_stream': {
+            hidden: true,
+          },
+        },
+        isIndexModeTimeSeries: false,
+      });
+      expect(templateWithDataStreamHidden.data_stream.hidden).toEqual(true);
+
+      const templateWithoutDataStreamHidden = getTemplate({
+        templateIndexPattern,
+        type: 'logs',
+        packageName: 'nginx',
+        composedOfTemplates: [],
+        templatePriority: 200,
+        hidden: true,
+        isIndexModeTimeSeries: false,
+      });
+      expect(templateWithoutDataStreamHidden.data_stream.hidden).toEqual(true);
 
-    expect(mappings).toMatchSnapshot(path.basename(ymlPath));
-  });
+      const templateWithGlobalHiddenTrueAndDataStreamHiddenFalse = getTemplate({
+        templateIndexPattern,
+        type: 'logs',
+        packageName: 'nginx',
+        composedOfTemplates: [],
+        templatePriority: 200,
+        hidden: true,
+        registryElasticsearch: {
+          'index_template.data_stream': {
+            hidden: false,
+          },
+        },
+        isIndexModeTimeSeries: false,
+      });
+      expect(templateWithGlobalHiddenTrueAndDataStreamHiddenFalse.data_stream.hidden).toEqual(true);
+
+      const templateWithoutHidden = getTemplate({
+        templateIndexPattern,
+        type: 'logs',
+        packageName: 'nginx',
+        composedOfTemplates: [],
+        templatePriority: 200,
+        isIndexModeTimeSeries: false,
+      });
+      expect(templateWithoutHidden.data_stream.hidden).toEqual(undefined);
+    });
 
-  it('tests loading system.yml', () => {
-    const ymlPath = path.join(__dirname, '../../fields/tests/system.yml');
-    const fieldsYML = readFileSync(ymlPath, 'utf-8');
-    const fields: Field[] = load(fieldsYML);
-    const processedFields = processFields(fields);
+    it('adds composed_of correctly for otel input packages of type logs', () => {
+      const composedOfTemplates = ['component1', 'component2'];
 
-    const mappings = generateMappings(processedFields);
+      const template = getTemplate({
+        templateIndexPattern: 'logs.otel-*',
+        type: 'logs',
+        packageName: 'otel-test',
+        composedOfTemplates,
+        templatePriority: 200,
+        isOtelInputType: true,
+      });
+      expect(template.composed_of).toStrictEqual([
+        OTEL_COMPONENT_TEMPLATE_MAPPINGS,
+        OTEL_COMPONENT_TEMPLATE_SETTINGS,
+        OTEL_COMPONENT_TEMPLATE_LOGS_MAPPINGS,
+        OTEL_COMPONENT_SEMCONV_RESOURCE_TO_ECS_MAPPINGS,
+        OTEL_COMPONENT_TEMPLATE_LOGS_CUSTOM_MAPPINGS,
```

Comment:
All custom templates must be after the `@package` templates, so they can be used to override fields or settings configured in packages. Regarding order of custom templates, more general ones should be placed first, so they can be overridden by the more specific ones.